goroutine/gid_tinygo.go

```diff
@@ -0,0 +1,37 @@
+//go:build tinygo
+
+package goroutine
+
+import (
+	"sync"
+	"unsafe"
+)
+
+const IsTinyGo = true
+
+var (
+	counter int64
+	mapping = make(map[uintptr]int64)
+	// TinyGo at the moment is single threaded so this is not needed but it's good to have anyway
+	// in case that changes. It does had ~5ns (from 20ns vs 4ns big go) but it's better to be correct.
+	// In theory the mutex could be noop on platforms where everything is single threaded.
+	lock sync.Mutex
+)
+
+func ID() int64 {
+	task := uintptr(currentTask())
+	lock.Lock()
+	if id, ok := mapping[task]; ok {
+		lock.Unlock()
+		return id
+	}
+	counter++
+	mapping[task] = counter
+	lock.Unlock()
+	return counter
```

Comment:
```suggestion
func ID() int64 {
	task := uintptr(currentTask())
	lock.Lock()
	defer lock.Unlock()
	if id, ok := mapping[task]; ok {
		return id
	}
	counter++
	mapping[task] = counter
	return counter
```

Reply:
I don't like defer when the code is a few lines and the exit points are clear
in theory I could for instance log the value after the unlock and minimize the critical section

admittedly there is nothing outside of the critical section but the return itself and the defer version is one less line so... could be fine

I also considered that in tinygo maybe defer is more expensive for the compiler to implement than explicit

adding a comment